source/postprocess/point_values.cc

```diff
@@ -59,12 +59,21 @@ namespace aspect
             }
 
           // ensure that exactly one processor found things
-          AssertThrow (Utilities::MPI::sum (point_found ? 1 : 0, this->get_mpi_communicator()) == 1,
-                       ExcMessage ("While trying to evaluate the values of the solution at "
-                                   "evaluation point " + Utilities::int_to_string(p) +
-                                   ", no (or more than one) processor reported that that point lies inside the "
-                                   "set of cells it owns. Are you trying to evaluate the "
-                                   "solution at a point that lies outside the domain?"));
+          const int nfound = Utilities::MPI::sum (point_found ? 1 : 0, this->get_mpi_communicator());
+          AssertThrow (nfound == 1,
```

Comment:
I think you want to inline the call to `MPI::sum` into the assertion because then the sum is not formed in release mode.

If you want to reuse the result a second time, you could either write the call a second time into the code, or do something like
```
  int n_found;
  Assert ((n_found = ...) == 1,
              ... uses n_found...);
```